app/controllers/services_controller.rb

```diff
@@ -4,6 +4,7 @@ class ServicesController < ApplicationController
 
   def register
     changes_made = false
+    changed_service = nil
```

Comment:
This shouldn't be necessary. If `changes_made` is false anyway, this variable doesn't need to be defined.